checker/tests/regex/Issue3281.java

```diff
@@ -44,6 +44,7 @@ void m3(String s) {
     void m4(String s, String s2) {
         RegexUtil.isRegex(s);
         if (RegexUtil.isRegex(s2)) {
+            // :: error: (argument.type.incompatible)
             Pattern.compile(s);
         }
```

Comment:
Can you please also add these:

````
    void foo(String s1, String s2) {
        bar(RegexUtil.isRegex(s1), 
            // :: error: (argument.type.incompatible)
            Pattern.compile(s1));
        boolean b;
        bar(b = RegexUtil.isRegex(s2),
            // :: error: (argument.type.incompatible)
            Pattern.compile(s2));
    }

    void bar(boolean b, Object o) {}
````

They currently do the right thing, but making sure that the conditional stores don't flow into following arguments would be good (in particular if we change the transfer function for assignment).